src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private MyNode<T> first;
+    private int size;
 
+    public void push(T value) {
+        MyNode<T> currentNode = new MyNode<>();
+        if (size == 0) {
+            first = currentNode;
+            currentNode.next = null;
+        } else {
+            currentNode.next = first;
+            first = currentNode;
+        }
+        currentNode.value = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        checkStack();
+        return first.value;
     }
 
     public T pop() {
-        return null;
+        checkStack();
+        T currentNodeValue = first.value;
+        first = first.next;
+        size--;
+        return currentNodeValue;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private void checkStack() {
+        if (size == 0) {
+            throw new EmptyStackException();
+        }
+    }
+
+    private static class MyNode<T> {
+        T value;
+        MyNode<T> next;
```

Comment:
add access modifier and let's create a constructor for two fields
you will be able to use this constructor in `push` method